tests/integration/targets/ec2_lc/tasks/main.yml

```diff
@@ -0,0 +1,177 @@
+        assign_public_ip: yes
+        instance_type: '{{ ec2_instance_type }}'
+        security_groups: '{{ sg.group_id }}'
+        volumes:
+          - device_name: /dev/xvda
+            volume_size: 10
+            volume_type: gp2
+            delete_on_termination: true
+      register: lc_1_create
+
+    - assert:
+        that:
+          - lc_1_create is changed
+          - '"autoscaling:CreateLaunchConfiguration" in lc_1_create.resource_actions'
+
+# Create lc 1 - idempotency
+    - name: Create launch configuration - Idempotency
+      community.aws.ec2_lc:
+        name: '{{ resource_prefix }}-lc1'
+        image_id: '{{ ec2_ami_id }}'
+        assign_public_ip: yes
+        instance_type: '{{ ec2_instance_type }}'
+        security_groups: '{{ sg.group_id }}'
+        volumes:
+          - device_name: /dev/xvda
+            volume_size: 10
+            volume_type: gp2
+            delete_on_termination: true
+      register: lc_1_create_idem
+
+    - assert:
+        that:
+          - lc_1_create_idem is not changed
+          - '"autoscaling:CreateLaunchConfiguration" not in lc_1_create_idem.resource_actions'
+
+# Create lc 2
+    - name: Create launch configuration 2
+      community.aws.ec2_lc:
+        name: '{{ resource_prefix }}-lc2'
+        image_id: '{{ ec2_ami_id }}'
+        assign_public_ip: yes
+        instance_type: '{{ ec2_instance_type }}'
+        security_groups: '{{ sg.group_id }}'
+        volumes:
+          - device_name: /dev/xvda
+            volume_size: 10
+            volume_type: gp2
+            delete_on_termination: true
+      register: lc_2_create
+
+    - assert:
+        that:
+          - lc_2_create is changed
+          - '"autoscaling:CreateLaunchConfiguration" in lc_2_create.resource_actions'
+
+# Create lc 2 - idempotency
+    - name: Create launch configuration 2 - Idempotency
+      community.aws.ec2_lc:
+        name: '{{ resource_prefix }}-lc2'
+        image_id: '{{ ec2_ami_id }}'
+        assign_public_ip: yes
+        instance_type: '{{ ec2_instance_type }}'
+        security_groups: '{{ sg.group_id }}'
+        volumes:
+          - device_name: /dev/xvda
+            volume_size: 10
+            volume_type: gp2
+            delete_on_termination: true
+      register: lc_2_create_idem
+
+    - assert:
+        that:
+          - lc_2_create_idem is not changed
+          - '"autoscaling:CreateLaunchConfiguration" not in lc_2_create_idem.resource_actions'
+
+# Find aws launch configuration
+    - name: Search for the Launch Configurations that start with test resource_prefix
+      community.aws.ec2_lc_find:
+        name_regex: '{{ resource_prefix }}*'
+        sort_order: descending
+        limit: 2
+      register: lc_find_result
+
+    - assert:
+        that:
+          - lc_find_result.results | length == 2
+          - '"autoscaling:DescribeLaunchConfigurations" in lc_find_result.resource_actions'
+
+    - debug: msg="{{ lc_find_result }}"
+
+  always:
```

Comment:
I'd remove the `always` block here. As it is now, assuming the `- name: Create launch configuration - Idempotency` task fails, it will automatically go into the `always` block. Here, `- name: Delete launch configuration 1` fails since `- name: Create launch configuration - Idempotency` was unsuccessful l and it won't release all the other resources you created (vpc, subnets, etc).

I would suggest leaving all these tasks outside of the always block (we want to test if the deletion works correctly), and at the end, add the always block deleting all the resources (you don't need to assert anything here).

Reply:
Hi @alinabuzachis , I'm not sure if I'm missing something, because ```name: Create launch configuration - Idempotency``` seems to be working fine.

```
TASK [ec2_lc : Create launch configuration 1 - Idempotency] ***************************************************************************************************
task path: /root/ansible_collections/community/aws/tests/output/.tmp/integration/ec2_lc-0tll458d-ÅÑŚÌβŁÈ/tests/integration/targets/ec2_lc/tasks/main.yml:35
Using module file /root/ansible_collections/community/aws/plugins/modules/ec2_lc.py
Pipelining is enabled.
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<testhost> EXEC /bin/sh -c 'ANSIBLE_DEBUG_BOTOCORE_LOGS=True /usr/bin/python3.9 && sleep 0'
ok: [testhost] => {
    "arn": null,
    "changed": false,
    "created_time": "None",
    "image_id": "ami-0d0602b784b8d07a5",
    "instance_type": "t2.micro",
    "invocation": {
        "module_args": {
            "assign_public_ip": true,
            "associate_public_ip_address": null,
            "aws_access_key": "ASIA6CCDWXDOAIDBG4EV",
            "aws_ca_bundle": null,
            "aws_config": null,
            "aws_secret_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "classic_link_vpc_id": null,
            "classic_link_vpc_security_groups": null,
            "debug_botocore_endpoint_logs": true,
            "ebs_optimized": false,
            "ec2_url": null,
            "image_id": "ami-0d0602b784b8d07a5",
            "instance_id": null,
            "instance_monitoring": false,
            "instance_profile_name": null,
            "instance_type": "t2.micro",
            "kernel_id": null,
            "key_name": null,
            "name": "ansible-test-63574971-localhost-lc1",
            "placement_tenancy": null,
            "profile": null,
            "ramdisk_id": null,
            "region": "us-east-1",
            "security_groups": [
                "sg-08411bbb760de2f19"
            ],
            "security_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "spot_price": null,
            "state": "present",
            "user_data": null,
            "user_data_path": null,
            "validate_certs": true,
            "volumes": [
                {
                    "delete_on_termination": true,
                    "device_name": "/dev/xvda",
                    "volume_size": 10,
                    "volume_type": "gp2"
                }
            ],
            "vpc_id": null
        }
    },
    "name": "ansible-test-63574971-localhost-lc1",
    "resource_actions": [
        "ec2:DescribeSecurityGroups",
        "autoscaling:DescribeLaunchConfigurations"
    ],
    "result": {
        "associate_public_ip_address": true,
        "block_device_mappings": [
            {
                "device_name": "/dev/xvda",
                "ebs": {
                    "snapshot_id": null,
                    "volume_size": 10
                },
                "virtual_name": null
            }
        ],
        "created_time": "None",
        "ebs_optimized": false,
        "image_id": "ami-0d0602b784b8d07a5",
        "instance_monitoring": false,
        "instance_type": "t2.micro",
        "launch_configuration_name": "ansible-test-63574971-localhost-lc1",
        "security_groups": [
            "sg-08411bbb760de2f19"
        ]
    },
    "security_groups": [
        "sg-08411bbb760de2f19"
    ]
}

TASK [ec2_lc : assert] ****************************************************************************************************************************************
task path: /root/ansible_collections/community/aws/tests/output/.tmp/integration/ec2_lc-0tll458d-ÅÑŚÌβŁÈ/tests/integration/targets/ec2_lc/tasks/main.yml:49
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

```

Reply:
@mandar242 I only gave an example. I you try to have `- name: Create launch configuration - Idempotency` fail, you won't release all the resources you have created. This was the purpose of the example.

Reply:
@mandar242 let me know if this makes sense for you now.